netmaster/master/netmaster_test.go

```diff
@@ -296,3 +308,164 @@ func TestVxlanConfigWithLateHostBindings(t *testing.T) {
+            "PktTagType"        : "vxlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, true)
+
+	CfgBytes = []byte(`{
+    "Tenants" : [{
+        "Name"                  : "tenant5",
+        "DefaultNetType"        : "vxlan",
+        "SubnetPool"            : "11.1.0.0/16",
+        "AllocSubnetLen"        : 24,
+        "Vlans"                 : "1201-1500",
+        "Vxlans"                : "2001-3000",
+        "Networks"  : [{
+            "Name"              : "net5",
+            "PktTag"            : 1200,
+            "PktTagType"        : "vlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, true)
+
+	CfgBytes = []byte(`{
+    "Tenants" : [{
+        "Name"                  : "tenant6",
+        "DefaultNetType"        : "vxlan",
+        "SubnetPool"            : "11.1.0.0/16",
+        "AllocSubnetLen"        : 24,
+        "Vlans"                 : "1201-1500",
+        "Vxlans"                : "2001-3000",
+        "Networks"  : [{
+            "Name"              : "net6",
+            "PktTag"            : 1201,
+            "PktTagType"        : "vlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, false)
+
+	CfgBytes = []byte(`{
+    "Tenants" : [{
+        "Name"                  : "tenant7",
+        "DefaultNetType"        : "vxlan",
+        "SubnetPool"            : "11.1.0.0/16",
+        "AllocSubnetLen"        : 24,
+        "Vlans"                 : "1201-1500",
+        "Vxlans"                : "2001-3000",
+        "Networks"  : [{
+            "Name"              : "net7",
+            "PktTag"            : 1500,
+            "PktTagType"        : "vlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, false)
+
+	CfgBytes = []byte(`{
+    "Tenants" : [{
+        "Name"                  : "tenant8",
+        "DefaultNetType"        : "vxlan",
+        "SubnetPool"            : "11.1.0.0/16",
+        "AllocSubnetLen"        : 24,
+        "Vlans"                 : "1201-1500",
+        "Vxlans"                : "2001-3000",
+        "Networks"  : [{
+            "Name"              : "net8",
+            "PktTag"            : 1501,
+            "PktTagType"        : "vlan"
+        }]
+    }]}`)
+	applyVerifyRangeTag(t, CfgBytes, true)
+
+}
+
+func applyVerifyRangeTag(t *testing.T, cfgBytes []byte, shouldFail bool) {
+	initFakeStateDriver(t)
+	defer deinitFakeStateDriver()
+
+	cfg := &intent.Config{}
+	err := json.Unmarshal(cfgBytes, cfg)
+	if err != nil {
+		t.Fatalf("error '%s' parsing config '%s'\n", err, cfgBytes)
+	}
+
+	_, err = resources.NewStateResourceManager(fakeDriver)
+	if err != nil {
+		log.Fatalf("state store initialization failed. Error: %s", err)
+	}
+	defer func() { resources.ReleaseStateResourceManager() }()
+
+	tenant := cfg.Tenants[0]
+	err = CreateTenant(fakeDriver, &tenant)
+	if err != nil {
+		t.Fatalf("error '%s' creating tenant\n", err)
+	}
+
+	err = CreateNetworks(fakeDriver, &tenant)
+	if shouldFail {
+		if err == nil {
```

Comment:
it might be good to test that err returned is indeed for `OutOfRange` as `CreateNetworks()` might return other errors as well?